src/psbt/mod.rs

```diff
@@ -37,15 +48,39 @@ impl PsbtUtils for Psbt {
             None
         }
     }
+
+    fn fee_amount(&self) -> Option<u64> {
+        let tx = &self.unsigned_tx;
+        let utxos: Option<Vec<TxOut>> = (0..tx.input.len()).map(|i| self.get_utxo_for(i)).collect();
+
+        utxos.map(|inputs| {
+            let input_amount = inputs.iter().fold(0, |acc, i| acc + i.value);
+
+            let output_amount = &self
+                .unsigned_tx
+                .output
+                .iter()
+                .fold(0, |acc, o| acc + o.value);
+
+            input_amount - output_amount
+        })
+    }
+
+    fn fee_rate(&self) -> Option<FeeRate> {
+        let fee_amount = self.fee_amount();
+        let weight = self.clone().extract_tx().weight();
```

Comment:
It's a small nit, but I would move this line inside the closure so that you end up cloning the whole PSBT only if `fee_amount` is `Some()`. Otherwise it's just wasted time, because you'll throw away the weight immediately

Reply:
Oh ya good idea, fixed.